pkg/controller/runtime/internal/qruntime/internal/containers/priority_queue_test.go

```diff
@@ -110,12 +110,12 @@ func TestPriorityQueueDeduplicateSimple(t *testing.T) {
 func TestPriorityQueueDeduplicateWithReleaseAfter(t *testing.T) {
 	var q containers.PriorityQueue[int]
 
+	var base time.Time
+
 	for _, i := range []int{1, 2, 3, 4, 5} {
-		assert.True(t, q.Push(i, time.Time{}))
+		assert.True(t, q.Push(i, base.Add(time.Duration(20-i))))
```

Comment:
these are different elements, something is broken if this needs to change

Reply:
I add some items with time greater than the time in the next loop. That's the test for the new logic.
But I've actually expanded the tests now, so that it checks both adding the items with greater timestamps and smaller ones.